### PR TITLE
crowbar: Add rake task "schema_migrate_status" for migration overview

### PR DESCRIPTION
**Pull request by toabctl**
To get an overview about the current migration status for the different
proposals, add a new rake task called "schema_migrate_status".
Calling the task with

RAILS_ENV=production rake crowbar:schema_migrate_status

gives a list with the template name, the latest possible revision and a list
of proposals with the currently applied revision. It looks like:

```
*barclamp*           *latest revision*    *proposals revision*
nova                 101                  default:101
keystone             35                   default:35
rabbitmq             11                   default:11
neutron              48                   default:48
```

### CODE REVIEW

**Comment by toabctl** [comment]
Not sure if this is useful. I just wanted to learn a bit about crowbar and the migrations....

File: crowbar_framework/lib/tasks/crowbar.rake [hunk 1]
```diff
@@ -37,4 +37,29 @@ namespace :crowbar do
...
+    end
+
+    printf "%-20s %-20s %s\n", "*barclamp*", "*latest revision*", "*proposals revision*"
+    barclamps.split.each do |bc_name|
+      latest_schema_revision, latest_proposals_revision = \
+                              SchemaMigration.get_barclamp_current_deployment_revison bc_name
+      unless latest_proposals_revision.nil?
+        printf "%-20s %-20s %s\n", bc_name, latest_schema_revision,
+               latest_proposals_revision.collect{
+          |prop| "#{prop[:name]}:#{prop[:revision]}"}.join(" ")
```
Comment (vuntz) [inline crowbar_framework/lib/tasks/crowbar.rake]: Interesting that hound doesn't complain. Correct style would be:
```ruby
unless latest_proposals_revision.nil?
  proposals_rev = latest_proposals_revision.collect { |prop| "#{prop[:name]}:#{prop[:revision]}" }
  printf "%-20s %-20s %s\n", bc_name, latest_schema_revision, proposals_rev.join(" ")
end
```

File: crowbar_framework/lib/tasks/crowbar.rake [hunk 2]
```diff
@@ -37,4 +37,29 @@ namespace :crowbar do
...
+    require "barclamp_catalog"
+
+    if args[:barclamps].nil?
+      barclamps = BarclampCatalog.barclamps.keys.join(" ")
+    else
+      barclamps = args[:barclamps]
+    end
+
+    printf "%-20s %-20s %s\n", "*barclamp*", "*latest revision*", "*proposals revision*"
+    barclamps.split.each do |bc_name|
```
Comment (vuntz) [inline crowbar_framework/lib/tasks/crowbar.rake]: Can you sort barclamps by name?

File: crowbar_framework/lib/schema_migration.rb [hunk 3]
```diff
@@ -79,6 +79,21 @@ def self.migrate_proposal_from_json(bc_name, json)
...
+      template = Proposal.new(barclamp: bc_name)
+    rescue Proposal::TemplateMissing
+      return [nil, nil]
+    end
+
+    proposals = Proposal.where(barclamp: bc_name)
+    latest_schema_revision = template["deployment"][bc_name]["schema-revision"]
+    latest_proposals_revision = proposals.collect do |prop|
+      { name: prop.name, revision: prop["deployment"][bc_name]["schema-revision"] }
+    end
```
Comment (vuntz) [inline crowbar_framework/lib/schema_migration.rb]: Can you sort this by proposal name?

**Comment by vuntz** [comment]
> Not sure if this is useful.

Feels useful to me :-)

File: crowbar_framework/lib/schema_migration.rb (hunk 3 above)
Comment (toabctl) [inline crowbar_framework/lib/schema_migration.rb]: done

File: crowbar_framework/lib/tasks/crowbar.rake (hunk 2 above)
Comment (toabctl) [inline crowbar_framework/lib/tasks/crowbar.rake]: done

File: crowbar_framework/lib/tasks/crowbar.rake (hunk 1 above)
Comment (toabctl) [inline crowbar_framework/lib/tasks/crowbar.rake]: done

**Comment by toabctl** [comment]
@vuntz fixed you comments. and added another cleanup commit.

**Comment by vuntz** [comment]
+1

**Comment by rsalevsky** [comment]
+1

If you want to make the output a little bit better then you can use https://github.com/tj/terminal-table. We use it anyway already in the crowbar-client.

**Comment by toabctl** [comment]
> If you want to make the output a little bit better then you can use https://github.com/tj/terminal-table. We use it anyway already in the crowbar-client.

Imo not needed. And keeping the requirements for the server at a minimum is a good thing.

**Comment by aspiers** [comment]
Nice addition, thanks :)

**Pull request merged**